src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs

```diff
@@ -50,5 +54,32 @@ public override bool IsLazyEvalParam(int index)
         {
             return LamdaParamMask.TestBit(index);
         }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+
+            var isValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            // Check if all record names of args exist against return type and if its possible to coerce.
+            if (isValid && returnType.IsRecord)
+            {
+                for (var i = 0; i < args.Length; i++)
+                {
+                    DType curType = argTypes[i];
+
+                    if (curType.IsRecord && !curType.CheckAggregateNames(returnType, args[i], errors, SupportsParamCoercion, context.Features.PowerFxV1CompatibilityRules))
```

Comment:
>returnType [](http://example.com/codeflow?start=73&length=10)

Should we check it against paramTypes? Instead of return type.

Reply:
Check returntype here because compare between argtypes and return type. paramtypes is argTypes so cannot use here.

Reply:
I see, i updated to use paramtypes here